addons/stock_account/models/product.py

```diff
@@ -91,6 +91,9 @@ def write(self, vals):
         res = super(ProductTemplate, self).write(vals)
 
         for product_template, (products, description, products_orig_quantity_svl) in impacted_templates.items():
+            if not products.exists():
```

Comment:
While it is true that if you didn't have units in stock adding attribute values and hence asking the system to create new variants linked to the product will delete the already existing variant during the above write:
https://github.com/odoo/odoo/blob/3c4275fb00255e519f01bf5547eff1db3a59d4b5/addons/stock_account/models/product.py#L91-L95
If you had unis in stock the record will be archived instead and the existing quant will just refer to the archived product for the given quantity. 
In particular, if products do not exist, I think it would be best to just continue since there is nothing to empty (there is no stock in the first place).

```suggestion
if not products.exists():
   continue
```